src/EmailAddressValidator.php

```diff
@@ -0,0 +1,295 @@
+        if (preg_match('/^[' . $atext . ']+(\x2e+[' . $atext . ']+)*$/', $localPart)) {
+            $result = true;
+        } else {
+            // Try quoted string format (RFC 5321 Chapter 4.1.2)
+
+            // Quoted-string characters are: DQUOTE *(qtext/quoted-pair) DQUOTE
+            $qtext = '\x20-\x21\x23-\x5b\x5d-\x7e'; // %d32-33 / %d35-91 / %d93-126
+            $quotedPair = '\x20-\x7e'; // %d92 %d32-126
+            if (preg_match('/^"([' . $qtext . ']|\x5c[' . $quotedPair . '])*"$/', $localPart)) {
+                $result = true;
+            } else {
+                $this->addMessage(sprintf(
+                    $this->translate("'%s' can not be matched against dot-atom format"),
+                    $localPart
+                ));
+                $this->addMessage(sprintf(
+                    $this->translate("'%s' can not be matched against quoted-string format"),
+                    $localPart
+                ));
+                $this->addMessage(sprintf(
+                    $this->translate("'%s' is not a valid local part for email address '%s'"),
+                    $localPart,
+                    $email
+                ));
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * Internal method to validate the hostname part of the email address
+     *
+     * @return boolean
+     */
+    private function validateHostnamePart($hostname, $email)
+    {
+        $hostValidator = new HostnameValidator();
+
+        if ($this->validateIp($hostname)) {
+            return $hostname;
+        }
+
+        if (preg_match('/^\[([^\]]*)\]$/i', $hostname, $matches)) {
+            $validHostname = $matches[1];
+            if (! $this->validateIp($validHostname)) {
+                $this->addMessage(sprintf(
+                    $this->translate("host name %s is a domain literal and is invalid"),
+                    $hostname
+                ));
+
+                return false;
+            }
+
+            return $validHostname;
+        }
+
+        if (! $hostValidator->isValid($hostname)) {
+            $this->addMessage(sprintf(
+                $this->translate('%s is not a valid domain name for email address %s.'),
+                $hostname,
+                $email
+            ));
+
+            return false;
+        } elseif ($this->mx) {
+            // MX check on hostname
+            $hostname = $this->validateMXRecords($hostname, $email);
+        } else {
+            $hostname = idn_to_ascii($hostname);
+        }
+
+        return $hostname;
+    }
+
+    public function validateIp($value)
+    {
+        if (! filter_var($value, FILTER_VALIDATE_IP)) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Returns true if and only if $value is a valid email address
+     * according to RFC2822
+     *
+     * @link   http://www.ietf.org/rfc/rfc2822.txt RFC2822
+     * @link   http://www.columbia.edu/kermit/ascii.html US-ASCII characters
```

Comment:
Please remove those links. The RFC is mentioned in the description. The other link.. is bogus, or, no idea, just a random ascii cheat sheet...